docs/ops/generation/RandomUniform_8.md

```diff
@@ -0,0 +1,195 @@
+    * **Range of values**: positive integers
+    * **Type**: `int`
+    * **Default value**: None
+    * **Required**: *Yes*
+
+**Inputs**:
+
+*   **1**: `shape` - 1D tensor of type *T_SHAPE* describing output shape. **Required.**
+
+*   **2**: `minval` - Scalar with type specified by the attribute *output_type*, defines the lower bound on the range of random values to generate (inclusive). Default value = 0.
+
+*   **3**: `maxval` - Scalar with type specified by the attribute *output_type*, defines the upper bound on the range of random values to generate (exclusive). Required for int64 and int32. Default value = 1.
+
+
+**Outputs**:
+
+* **1**: A tensor with type specified by the attribute *output_type* and shape defined by `shape` input tensor.
+
+**Types**
+
+* *T_SHAPE*: any supported integer type.
+
+
+**Detailed description**:
+
+*RandomUniform* operation generates random numbers from a uniform distribution in the range [*minval*, *maxval*). The generation algorithm is based on underlying random integer generator that uses Philox algorithm. Philox algorithm is counter based pseudo random generator, which produces unit32 values. Single invocation of Philox algorithm returns four result random values, depending on the given *key* and *counter* values. *Key* and *counter* are initialized with *seed* and *seed2* attributes respectively.
+
+The result of Philox is calculated by applying a fixed number of *key* and *counter* updating, so-called "rounds". This implementation uses 4x32_10 version of Philox algorithm, where number of rounds = 10.
+
+Suppose we have *n* which determines *n*-th 4 elements of random sequence.
+In each round *key*, *counter* and *n* are splitted to pairs of uint32 values:
+
+\f[
+R = cast\_to\_uint32(value)\\
+L = cast\_to\_uint32(value >> 32),
+\f]
+where *cast\_to\_uint32* - static cast to uint32, *value* - uint64 input value, *L*, *R* - uint32 result values, >> - bitwise right shift.
+
+Then *n* and *counter* are updated with the following formula:
+
+\f[
+L'= mullo(R, M)\\
+R' = mulhi(R, M) {\oplus} k {\oplus} L \\
+mulhi(a, b) = floor((a {\times} b) / 2^{32}) \\
+mullo(a, b) = (a {\times} b) \mod 2^{32}
+\f]
+where *k* = `R_{key}` for updating counter, *k* = `L_{key}` for updating *n*, *M* = `0xD2511F53` for updating *n*, *M* = `0xCD9E8D57` for updating *counter*.
+
+After each round *key* is raised by summing with another pair of const values:
+\f[
+L += 0x9E3779B9 \\
+R += 0xBB67AE85
+\f]
+Values *L'_{n}*, *R'_{n}*, *L'_{counter}*, *R'_{counter}* are resulting four random numbers.
+
+All const values are obtained from original paper:
+https://www.thesalmons.org/john/random123/papers/random123sc11.pdf
+
+Float values between [0..1) are obtained from 32-bit integers by the following rules.
+
+Float32 is formatted as follows: sign(1 bit) exponent(8 bits) mantissa(23 bits), so each element will have:
+``` 
+sign = 0
+exponent = 127 - representation of a zero exponent.
```

Comment:
Please, write more detailed description for the conversion to `fp16`, `fp32`, `fp64`.

Reply:
Extended description of obtaining fp32. Calculations for fp16 and fp64 are similar.

Reply:
Added description of obtaining fp16 and fp64.